src/app/model.ts

```diff
@@ -1,36 +1,161 @@
-import {Tickets, TicketType, Priority} from 'node-zendesk';
+import {Tickets} from 'node-zendesk';
 
-import {configStore} from '../store';
+import {AppFormValue, AppFormValues} from 'mattermost-redux/types/apps';
 
-type CreateTicketFormValues = {
-    subject: string;
-    type: TicketType;
-    priority: Priority;
-    additional_message: string;
-    post_message: string;
-}
+import {AppFieldNames, ZDFieldValidation, getMultiselectValues} from '../utils';
+
+import {configStore} from '../store';
 
-export const getTicketFromForm = (values: CreateTicketFormValues): Tickets.CreatePayload => {
-    const mmSignature = '*message created from Mattermost message.*\n' + configStore.getSiteURL();
+export type FieldValidationErrors = {[name: string]: string};
 
-    const additionalMessage = values.additional_message || '';
-    const postMessage = values.post_message || '';
+interface ITicketFromFrom {
+    getTicket(): Tickets.CreatePayload;
+    fieldValidationErrors: FieldValidationErrors;
+}
 
-    const zdMessage = additionalMessage + '\n' +
-            postMessage + '\n' +
-            mmSignature;
+export class TicketFromForm implements ITicketFromFrom {
+    formValues: AppFormValues;
+    ticket: Tickets.CreateModel;
+    fieldValidationErrors: FieldValidationErrors
 
-    const ticket: Tickets.CreatePayload = {
-        ticket: {
-            subject: values.subject,
-            type: values.type,
-            priority: values.priority,
+    constructor(values: AppFormValues) {
+        this.formValues = values;
+        this.fieldValidationErrors = {};
+        this.ticket = {
             comment: {
-                body: zdMessage,
+                body: this.getPostMessage(),
             },
-        },
-    };
-    return ticket;
-};
+        };
+    }
+
+    getPostMessage(): string {
+        const mmSignature = '*message created from Mattermost message.*\n' + configStore.getSiteURL();
+
+        const additionalMessage = this.formValues[AppFieldNames.AdditionalMessage] || '';
+        const postMessage = this.formValues[AppFieldNames.PostMessage] || '';
+
+        const zdMessage = additionalMessage + '\n' +
+                postMessage + '\n' +
+                mmSignature;
+
+        return zdMessage;
+    }
+
+    getTicket(): Tickets.CreatePayload {
+        this.mapFormValuesToTicket();
+        return {ticket: this.ticket};
+    }
+
+    // mapFormValuesToTicket maps AppCall values 1:1 to a zendesk ticket
+    mapFormValuesToTicket(): void {
+        // get only the fields that are to be mapped
+        const prunedFormValues = this.pruneFormFields();
+
+        // iterate through each field and build the Zendesk ticket
+        Object.keys(prunedFormValues).forEach((fieldName) => {
+            switch (true) {
```

Comment:
Unless we plan to handle other types of fields, removing the switch and using an `if` instead would reduce # of lines

Reply:
Right.  When I introduced the pruning method, this case statement was reduced in half and isn't really necessary anymore.  Done.